addon/globalPlugins/webAccess/gui/properties.py

```diff
@@ -26,1011 +26,1031 @@
-        self.gesturesOptions = []
-
-        # Instanciation of increment the values for wx.choice and updates the correspondant wx.listCtrl
-        self.objIncAutoAct = IncrDecrListPos()
-        self.objIncMut = IncrDecrListPos()
-        self.objIncGest = IncrDecrListPos()
-
-
-    def initData(self, context):
-        """
-        Init wx.listCtrl elements
-        """
-        self.context = context
-        self.toggleBtn.Disable()
-        self.editable.Disable()
-        self.choice.Disable()
-        self.getAutoActions()
-        self.getMutationOptions()
-        self.getGestureOptions()
-        
-
-    def setColumunWidthAutoSize(self):
-        for i in range(self.listCtrl.GetColumnCount()):
-            self.listCtrl.SetColumnWidth(i, wx.LIST_AUTOSIZE_USEHEADER)
-
-    def showItems(self, data):
-        ruleType = data.get("type")
-        typeValues = RULE_TYPE_FIELDS.get(ruleType)
-        if typeValues is None:
-            for item in self.hidable:
-                item.Hide()
-            self.noPropertiesLabel.Show()
-        else:
-            for item in self.hidable:
-                item.Show()
-            self.noPropertiesLabel.Hide()
-
-    def initPropertiesList(self, context):
-        self.context = context
-        index=self.listCtrl.GetFirstSelected()
-        propsList = ListProperties()
-        propsList.setPropertiesByRuleType(self.context)
-        self.propertiesList = propsList.getPropertiesByRuleType()
-        self.focusListCtrl(index=index)
-
-
-    def updateListCtrl(self, data):
-        """SU ruleProps:
-            data = dataRule["properties"]"""
-        if data:
-            for props in self.propertiesList:
-                for key, value in data.items():
-                    if props.get_id() == key:
-                        props.set_flag(True)
-                        props.set_value(value)
-
-    @staticmethod
-    def getAltFieldLabel(ruleType, key, default=None):
-        """
-        Function defines as static to customize customvalue and customName properties
-        according to the rule
-        """
-        if key == "customValue":
-            if ruleType in (ruleTypes.PAGE_TITLE_1, ruleTypes.PAGE_TITLE_2):
-                # Translator: Field label on the RulePropertiesEditor dialog.
-                return _("Custom page title")
-            elif ruleType in (ruleTypes.ZONE, ruleTypes.MARKER):
-                # Translator: Field label on the RulePropertiesEditor dialog.
-                return _("Custom message")
-        elif key == "customName" and ruleType in (ruleTypes.ZONE, ruleTypes.MARKER):
-            # Translator: Field label on the RulePropertiesEditor dialog.
-            return _("Custom name")
-        return default
-
-
-    def getKeyByRuleType(self, displayName):
-        """
-        Function converts the appropriate display name to the listctrl activate funtions
-        """
-        if displayName == "Custom page title" or displayName == "Custom message":
```

Comment:
I'm afraid this won't work once the UI is translated.
It could if the comparison literals were first passed through gettext.
However, there is no point in comparing display names as the rows of the list cannot be reordered and properties are known and in definite order for each rule type.